task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -754,6 +754,12 @@ def finalize(ti: RuntimeTaskInstance, state: TerminalTIState, log: Logger):
         )
         # TODO: Run task failure callbacks here
 
+    # Pushing xcom for each operator extra links defined
+    for oe in ti.task.operator_extra_link_dict.values():
+        link, xcom_key = oe.get_link(operator=ti.task, ti_key=ti.id), oe.xcom_key  # type: ignore[arg-type]
+        log.debug("Setting xcom for operator extra link", link=link, xcom_key=xcom_key)
+        _xcom_push(ti, key=xcom_key, value=link)
```

Comment:
Nit: lets move this to before the listeners (they _could_ timeout, and this should be quick and happen before hand)

Reply:
Done in last commit: a2a0cc559fee08f03fbcdc6b4b5c96b0171b3656